packages/remix-react/components.tsx

```diff
@@ -1573,22 +1587,29 @@ function convertRouterFetcherToRemixFetcher(
     formMethod &&
     formAction &&
     formEncType &&
-    formData
+    (formData || json || text)
```

Comment:
do we want both `json !== undefined` && `text != null` here? This would block null and empty string JSON submission, and empty string text submissions. If the "json" value here is pre-encoded (a json string) that one is fine, but the text case I think would cause issues.